pyresample/test/test_sgeom/test_arc.py

```diff
@@ -0,0 +1,312 @@
+        start_vertices, end_vertices = arc.vertices_in_degrees
+        assert np.allclose(start_vertices, start_point_vertices)
+        assert np.allclose(end_vertices, end_point_vertices)
+
+    def test_to_shapely(self):
+        """Test conversion to shapely."""
+        start_point_vertices = np.deg2rad((-180.0, 0.0))
+        end_point_vertices = np.deg2rad((-180.0, 90.0))
+        arc = SArc(SPoint.from_degrees(* start_point_vertices),
+                   SPoint.from_degrees(* end_point_vertices))
+        shapely_line = arc.to_shapely()
+        expected_line = LineString((start_point_vertices, end_point_vertices))
+        assert shapely_line.equals_exact(expected_line, tolerance=1e-10)
+
+    def test_hash(self):
+        """Test arc hash."""
+        arc = SArc(SPoint.from_degrees(-180.0, -90.0), SPoint.from_degrees(-180.0, 0.0))
+        assert hash(arc) == -3096892178517935054
+
+    def test_midpoint(self):
+        """Test arc midpoint."""
+        start_point_vertices = np.deg2rad((-180.0, 0.0))
+        end_point_vertices = np.deg2rad((-180.0, 90.0))
+        arc = SArc(SPoint(*start_point_vertices), SPoint(*end_point_vertices))
+        midpoint = arc.midpoint()
+        assert isinstance(midpoint, SPoint)
+        assert np.allclose(midpoint.vertices_in_degrees[0], (-180, 45))
+
+    def test_intersection_point(self):
+        """Test SArc(s) intersection point."""
+        arc1 = SArc(SPoint.from_degrees(0, 0), SPoint.from_degrees(10, 10))
+        arc2 = SArc(SPoint.from_degrees(0, 10), SPoint.from_degrees(10, 0))
+        lon, lat = arc1.intersection_point(arc2)
+
+        np.testing.assert_allclose(np.rad2deg(lon), 5)
+        assert np.rad2deg(lat).round(7) == round(5.0575148968282093, 7)
+
+        # Test when same SArc
+        arc1 = SArc(SPoint(0, 0), SPoint.from_degrees(10, 10))
+        assert (arc1.intersection_point(arc1) is None)
+
+        # Test intersecting SArc(s)
+        arc1 = SArc(SPoint.from_degrees(24.341215776575297, 44.987819588259327),
+                    SPoint.from_degrees(18.842727517611817, 46.512483610284178))
+        arc2 = SArc(SPoint.from_degrees(20.165961750361905, 46.177305385810541),
+                    SPoint.from_degrees(20.253297585831707, 50.935830837274324))
+        p = SPoint.from_degrees(20.165957021925202, 46.177022633103398)
+        assert arc1.intersection_point(arc2) == p
+        assert arc1.intersects(arc2)
+
+        # Test non-intersecting SArc(s)
+        arc1 = SArc(SPoint.from_degrees(-2.4982818108326734, 48.596644847869655),
+                    SPoint.from_degrees(-2.9571441235622835, 49.165688435261394))
+        arc2 = SArc(SPoint.from_degrees(-3.4976667413531688, 48.562704872921373),
+                    SPoint.from_degrees(-5.893976312685715, 48.445795283217116))
+        assert arc1.intersection_point(arc2) is None
+        assert not arc1.intersects(arc2)
+        assert not bool(None)  # this occurs within the intersects method
```

Comment:
These section comments (and the code) suggest these should all be separate tests. If they require more than an descriptive function name then a docstring can be added.